godot-macros/src/docs.rs

```diff
@@ -39,52 +39,72 @@ pub fn make_inherent_impl_docs(
     constants: &[ConstDefinition],
     signals: &[SignalDefinition],
 ) -> TokenStream {
+    /// Generates TokenStream containing field definitions for documented methods and documentation blocks for constants and signals.
     fn pieces(
         functions: &[FuncDefinition],
         signals: &[SignalDefinition],
         constants: &[ConstDefinition],
-    ) -> Option<TokenStream> {
+    ) -> TokenStream {
+        let to_tagged = |s: String, tag: &str| -> String {
+            if s.is_empty() {
+                s
+            } else {
+                format!("<{tag}>{s}</{tag}>")
+            }
+        };
+
+        let signals_block = to_tagged(
+            signals
+                .iter()
+                .filter_map(make_signal_docs)
+                .collect::<String>(),
+            "signals",
+        );
+        let constants_block = to_tagged(
+            constants
+                .iter()
+                .map(|ConstDefinition { raw_constant }| raw_constant)
+                .filter_map(make_constant_docs)
+                .collect::<String>(),
+            "constants",
+        );
+
         let methods = functions
             .iter()
-            .map(make_method_docs)
-            .collect::<Option<String>>()?;
-        let signals = signals
-            .iter()
-            .map(make_signal_docs)
-            .collect::<Option<String>>()?;
-        let constants = constants
-            .iter()
-            .map(|ConstDefinition { raw_constant: x }| x)
-            .map(make_constant_docs)
-            .collect::<Option<String>>()?;
-        let field_definition = quote! {
+            .filter_map(make_method_docs)
+            .collect::<String>();
+
+        let methods = if methods.is_empty() {
+            quote! { None }
+        } else {
+            quote! {#methods.into()}
```

Comment:
```suggestion
            quote! { #methods.into() }
```
Also, what does the `.into()` do here? Is it possible to make the conversion more explicit? Since this is non-local generated code, things can be quite hard to understand otherwise.

Reply:
The into was for `T => Some(T)` but it isnt necessary anymore and I put it into a review comment but it didnt actually get changed

Reply:
Let's write `Some(...)` then. 10 times clearer.

That's one thing I truly hate about `From`, it sometimes encourages obfuscation with zero benefits.

Reply:
I'll change it to `Some(…)`, I aimed to keep it consistent with earlier version